ipfs-cluster-ctl/main.go

```diff
@@ -447,8 +469,15 @@ func request(method, path string, body io.Reader, args ...string) *http.Response
 	checkErr("creating request", err)
 	r.WithContext(ctx)
 
+	if len(defaultUsername) != 0 {
+		r.SetBasicAuth(defaultUsername, defaultPassword)
+	}
+
 	client := &http.Client{Transport: defaultTransport}
 	resp, err := client.Do(r)
+	if err == nil && resp.StatusCode >= 400 {
```

Comment:
Now that this has been brought to my attention :)

I think the best place to detect error and set exit code in such case is the end of `formatResponse`. This ensures the received data/error is shown to the user in the desired format. Only then you can check resp.StatusCode and exit with 2 when it was not a successful request.

Reply:
Gotcha, yeah I was definitely planning on discussion/correction on this approach. And out of curiosity, why exit code 2 specifically? Is there a good rule of thumb for the first few exit codes that I can keep in mind?

Reply:
Nah, I said 2 because 1 is used for the rest of errors, so just to use a different one for errors which are from responses. It's completely arbitrary...

Reply:
Gotcha, good to know. We should document that somewhere for any users who might want to distinguish between to the two.